### PR TITLE
Add tests for module compression_heatpumps_and_chillers

### PR DESCRIPTION
**Pull request by jakob-wo**
1) This PR adds tests for the module `compression_heatpumps_and_chillers.py`.

It also includes two additional changes (I know, I should have done this in seperate PRs - sry!)
2) The PR fixes the docstring of a pre-calculation-function. Before it said the input argument had to be of type list but in fact it can be a pandas.Series as well. I added that information to the docstring.

3) The PR adds Exceptions to the module: It is checked whether the input arguments have the proper type and length.

### CODE REVIEW

File: src/oemof/thermal/compression_heatpumps_and_chillers.py [hunk 1]
```diff
@@ -66,7 +67,28 @@ def calc_cops(temp_high, temp_low, quality_grade, temp_threshold_icing=2,
 
 
     """
-    # Make both lists (temp_low and temp_high) have the same length and
+    # Check if input arguments have proper type and length
```
Comment (jnnr) [inline src/oemof/thermal/compression_heatpumps_and_chillers.py:71]: Would be good to insert spaces between the different if / elif / else statements for readability

File: src/oemof/thermal/compression_heatpumps_and_chillers.py [hunk 2]
```diff
@@ -40,9 +41,9 @@ def calc_cops(temp_high, temp_low, quality_grade, temp_threshold_icing=2,
 
     Parameters
     ----------
-    temp_high : list of numerical values
+    temp_high : list or pandas.Series of numerical values
```
Comment (jnnr) [inline src/oemof/thermal/compression_heatpumps_and_chillers.py:47]: If you want to allow for the new datatype, you should also write a test for that feature.

File: src/oemof/thermal/compression_heatpumps_and_chillers.py [hunk 3]
```diff
@@ -66,7 +67,28 @@ def calc_cops(temp_high, temp_low, quality_grade, temp_threshold_icing=2,
...
+    if type(temp_high) == pd.Series:
+        pass
+    elif not isinstance(temp_high, list):
+        raise TypeError('Argument temp_high is not of type list!')
+    if len(temp_high) != len(temp_low):
+        if (len(temp_high) != 1) and ((len(temp_low) != 1)):
+            raise IndexError('Arguments temp_low and '
+                             'temp_high have to be of same '
+                             'length or one has to be of length 1 !')
+    if factor_icing is not None and consider_icing is False:
```
Comment (jnnr) [inline src/oemof/thermal/compression_heatpumps_and_chillers.py]: To make this easier, I would suggest to drop `consider_icing` and keep only `factor_icing`. Then check if `factor_icing` is not None and include it in the calculations if that is the case.

File: src/oemof/thermal/compression_heatpumps_and_chillers.py [hunk 4]
```diff
@@ -66,7 +67,28 @@ def calc_cops(temp_high, temp_low, quality_grade, temp_threshold_icing=2,
 
 
     """
-    # Make both lists (temp_low and temp_high) have the same length and
+    # Check if input arguments have proper type and length
+    if type(temp_low) == pd.Series:
```
Comment (jnnr) [inline src/oemof/thermal/compression_heatpumps_and_chillers.py]:
```suggestion
    if not (isinstance(temp_low, (pd.Series, list))):
        raise TypeError('Argument temp_low is not of type list or pd.Series.')
```

**Review by jnnr — CHANGES_REQUESTED** [review]

**Comment by jnnr** [comment]
Thanks for the tests! Looks good, just some minor remarks.

**Comment by jakob-wo** [comment]
Good feedback! 
I like your ideas, implemented all requested changes and adjusted the examples to work without the argument 'consider_icing'.

File: src/oemof/thermal/compression_heatpumps_and_chillers.py [hunk 5]
```diff
@@ -98,9 +118,8 @@ def calc_cops(temp_high, temp_low, quality_grade, temp_threshold_icing=2,
                 if t_l >= temp_threshold_icing + 273.15:
                     cops = cops + [quality_grade * t_h / (t_h - t_l)]
         elif mode == "chiller":
-            # Combining 'consider_icing' and mode 'chiller' is not possible!
-            cops = None
-
+            raise ValueError('Argument factor ising has '
```
Comment (jnnr) [inline src/oemof/thermal/compression_heatpumps_and_chillers.py]:
```suggestion
            raise ValueError("Argument factor_icing has "
```

File: src/oemof/thermal/compression_heatpumps_and_chillers.py [hunk 6]
```diff
@@ -98,9 +118,8 @@ def calc_cops(temp_high, temp_low, quality_grade, temp_threshold_icing=2,
...
+                             'to be None for mode=chiller!')
```
Comment (jnnr) [inline src/oemof/thermal/compression_heatpumps_and_chillers.py]:
```suggestion
                             "to be None for mode='chiller'!")
```

File: src/oemof/thermal/compression_heatpumps_and_chillers.py [hunk 7]
```diff
@@ -139,6 +158,9 @@ def calc_max_Q_dot_chill(nominal_conditions, cops):
 
 
     """
+    if not isinstance(cops, list):
+        raise TypeError('Argument cops is not of type list!')
```
Comment (jnnr) [inline src/oemof/thermal/compression_heatpumps_and_chillers.py]:
```suggestion
        raise TypeError("Argument 'cops' is not of type list!")
```

**Comment by jnnr** [comment]
Thanks, looks good. Just a little suggestion in for the error messages. Apart from that, it could be merged.

In another PR, you could implement the functionality of giving back a pd.Series as result when passing one as input. But that can be dealt with separately.

**Review by jnnr — APPROVED** [review]

**Pull request merged**